### DISCUSSION TITLE
Use InstancedMesh for 3dTiles Meshes

### DISCUSSION THREAD

**Discussion by alexLuky** [iTowns/itowns#1979] [Ideas]
May be it would increase performance of rendering 3D tiles if we use InstancedMesh class like in basic three.js example? I think it can reduce the number of draw calls...
[https://threejs.org/examples/#webgl_instancing_performance](https://threejs.org/examples/#webgl_instancing_performance)

**Comment by jailln** [comment]
This could not be used to display b3dm tiles since they don't represent instanced geometries but it could be used to implement 3D Tiles i3dm tiles support.

**Reply by alexLuky** [reply]
@jailln 
I recomend you to investigate this lib. It uses intancedMeshes for b3dm tiles:
"...
Using InstancedTileLoader and InstancedOGC3DTile allows displaying the same Tileset at many different places with little impact on performance. Each Tileset is independent in terms of it's position, orientation and level of detail but each tile is created as an "InstancedMesh" giving much higher performance when displaying the same Tileset many times.
..."
[https://github.com/ebeaufay/threedtiles](https://github.com/ebeaufay/threedtiles)

**Comment by jailln** [comment]
Yes but it uses it to display the same tileset (in his case a spaceship) in different locations and not to display a tileset where each tile represents distinct data.
We could do the same for such usecases (a 3D tiles tileset that should be instanced in several places), but it is not the main use case for 3D tiles tilesets to my knowledge. Do you have such a use case?

Thanks for sharing this implementation, maybe we can use some of the performance improvements tips that we don't have yet.

**Reply by alexLuky** [reply]
We don't have exactly this kind of this case... but ok, now I understand your point, the issue with material and texture limitations.
